### PR TITLE
Move examples to separate folders

### PR DESCRIPTION
**Pull request by Hofer-Julian**

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Why advantage does it bring? We still have the same amount of entries in the source folder but now we have more levels.

**Comment by Hofer-Julian** [comment]
The advantage of the intermediate step is indeed limited. It is not true though, that we have the same amount of entries, you probably forgot about the ui files.

The next step would be to add one README per example.

I would then suggest to discuss within a follow-up issue:
1. If we really want to keep all examples
2. How we could structure them, so that especially newcomers know where to look.

**Comment by sdroege** [comment]
> The next step would be to add one README per example.

And to also add a screenshot to each UI example.

I think overall this makes going through the examples much more friendly and less confusing for users. There's not just a collection of source files but a proper overview and pre-example documentation, and everything belong to an example is in the same directory instead of having everything put together.

**Comment by Hofer-Julian** [comment]
I think CI will be fixed as soon as https://github.com/gtk-rs/gtk-rs/pulls is merged and I rebased.

**Comment by Hofer-Julian** [comment]
@sdroege you requested a review, this is still a a draft PR though.

One other thing, I don't really know how to resolve the gir conflict.

**Comment by sdroege** [comment]
> One other thing, I don't really know how to resolve the gir conflict.

Reset it to the version that is in master :)

> @sdroege you requested a review, this is still a a draft PR though.

It would still be good to get some opinions on it before it's finished IMHO

**Comment by Hofer-Julian** [comment]
> Reset it to the version that is in master :)

Could you please tell me the command?
The web does not seem to be very helpful here.

**Comment by sdroege** [comment]
Just rebasing on top of master should fix this already in your case.

File: examples/build_travis.sh [hunk 1]
```diff
@@ -1,42 +0,0 @@
-#!/bin/sh
```
Comment (GuillaumeGomez) [inline examples/build_travis.sh:1]: Good catch!

**Comment by GuillaumeGomez** [comment]
So, overall this seems nice. Now I have a big issue with adding images directly in this repository. It increases the size a lot for finally small gain. I like the idea of having screenshots though, so if someone has an idea on how to go around this issue...

**Comment by Hofer-Julian** [comment]
> So, overall this seems nice. Now I have a big issue with adding images directly in this repository. It increases the size a lot for finally small gain. I like the idea of having screenshots though, so if someone has an idea on how to go around this issue...

[Git-lfs](https://git-lfs.github.com/)

Not sure if adding images which don't change are a big problem though

**Comment by GuillaumeGomez** [comment]
On the repository definitely. I don't want to download images to update code. Also, unless stricly needed, we'd better stay as far away as possible from git-lfs.

**Comment by Hofer-Julian** [comment]
Host the images on a website and link them there.

I would like to mention that most screenshots were already there, so the repo size would only shrink if we change history (unless I misunderstand how git works)

**Comment by Hofer-Julian** [comment]
> Also, unless stricly needed, we'd better stay as far away as possible from git-lfs.

Wouldn't mind if you'd have the same policy for git submodules :P

**Comment by GuillaumeGomez** [comment]
> I would like to mention that most screenshots were already there, so the repo size would only shrink if we change history (unless I misunderstand how git works)

Damn, I didn't clean up things before merging repositories... T_T

git submodules are necessary evil here. :p

**Comment by Hofer-Julian** [comment]
Not sure why the regen check fails.
Hints would be very welcome :)

**Comment by Hofer-Julian** [comment]
Apart from the CI failure it is ready for review now.

**Comment by sdroege** [comment]
The CI failure should be gone if you rebase on top of latest master

**Comment by sdroege** [comment]
@GuillaumeGomez What do we do with the images? They're there already so let's keep them for now and get rid of them at a later time separately?

**Comment by GuillaumeGomez** [comment]
For the existing images, there is nothing to do unless rewriting the whole history. But I really don't know if this is a good idea to add new ones...

**Comment by sdroege** [comment]
Doesn't seem like much of a problem to me, they're small and not changing often.

**Comment by GuillaumeGomez** [comment]
30KB x the number of added images is definitely not small in a git repository. I really don't know what to do here...

**Comment by sdroege** [comment]
How much is it in sum that is added as part of this PR?

**Comment by GuillaumeGomez** [comment]
Around 88KB.

**Comment by sdroege** [comment]
Why are we even talking about this then?

**Comment by Hofer-Julian** [comment]
I rebased and squashed the commits now.

I hope that the CI is fine now and that we can move the binaries discussion to a separate issue.

**Review by sdroege — APPROVED** [review]

**Comment by Hofer-Julian** [comment]
regen check still fails :(

**Comment by Hofer-Julian** [comment]
If one of you knows a magic command which might fix the CI, please feel free to push to this branch.

**Comment by sdroege** [comment]
I assume this already fails before your PR then, i.e. master currently fails the CI. There's nothing in your PR here that should break it.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Just rebasing on master should be enough for the failure so that's not a problem. I'm really sad not having see the images before. Well, what's done is done. Thanks for moving the examples, I really like how they look now!

**Comment by Hofer-Julian** [comment]
> Thanks for moving the examples, I really like how they look now!

Glad to hear that you like it as well :)

**Pull request merged**